test/gtest/uct/ib/ud_base.cc

```diff
@@ -87,8 +87,22 @@ void ud_base_test::set_tx_win(entity *e, uct_ud_psn_t size)
 
 void ud_base_test::disable_async(entity *e) 
 {
-    ucs_async_remove_handler(iface(e)->async.timer_id, 1);
+    ucs_status_t status;
+    int event_fd;
+
+    /* at least one connection has to be initiated in order to disable
+     * timer-based async progress */
+    ASSERT_TRUE(iface(e)->async.timer_id != 0);
+    status = ucs_async_remove_handler(iface(e)->async.timer_id, 1);
+    ASSERT_UCS_OK(status);
+    iface(e)->async.timer_id = 0;
```

Comment:
maybe we keep the timer, but add some debug flag to iface to prevent the async flag from actually doing anything?

Reply:
do you mean add flags field to UD iface structure?

Reply:
how about relying on ud_iface::tx.timer_disable ?

Reply:
is it still needed, with iface->timer_disable?